contracts/feature-tests/basic-features/src/lib.rs

```diff
@@ -1227,6 +1258,123 @@ pub trait BasicFeatures {
 		self.crypto().verify_secp256k1(key, message, signature)
 	}
 
+	#[endpoint]
+	fn add_ec(
+		&self,
+		//curve: &EllipticCurve<Self::BigUint>,
+		field_order: Self::BigUint,
+		base_point_order: Self::BigUint,
+		eq_constant: Self::BigUint,
+		x_base_point: Self::BigUint,
+		y_base_point: Self::BigUint,
+		size_of_field: i32,
+		x_first_point: Self::BigUint,
+		y_first_point: Self::BigUint,
+		x_second_point: Self::BigUint,
+		y_second_point: Self::BigUint,
+	) -> (Self::BigUint, Self::BigUint) {
+		let curve = EllipticCurve::<Self::BigUint>::new(
+			field_order,
+			base_point_order,
+			eq_constant,
+			x_base_point,
+			y_base_point,
+			size_of_field,
+		);
+		self.crypto().add_ec(
+			&curve,
+			x_first_point,
+			y_first_point,
+			x_second_point,
+			y_second_point,
+		)
+	}
```

Comment:
why not create yourself an equation struct and pass just 1 argument to the function in stead of 11
start from here: https://refactoring.guru/introduce-parameter-object

Reply:
the people working with your features will thank you later!